savoirfairelinux/sous-chef
# Better automate deploy

### Changes proposed in this pull request:

* Automate the frontend compilation steps inside Dockerfile
* Make it so that `docker-compose up` is enough to get the project running (admittedly with an empty database.)
* Tidy up the subsidiary docker-compose files by coalescing their common bits directly into Dockerfile or the main docker-compose.yml
* Don't use docker-compose.override.yml, because explicit is better than implicit: instead name it docker-compose.dev.yml
* Add copious comments to the Dockerfile.

### Status

This was branched off #757 so please merge that first.

- [] READY

### How to verify this change

Do a fresh checkout and a fresh build. See that it works with just `docker-compose up`. Try then with `docker-compose -f docker-compose.yml -f docker-compose.yml up`

### Deployment notes and migration


### Additional notes

Deployment has been a huge headache for me. INSTALL.md is pretty nicely fleshed out, but it's still really hard to knowing when and what order to run things, and I would rather move as many instructions into (commented) scripts instead of into instructions that humans need to follow manually.
